### PR TITLE
health: Annotate k8s node with health IPs in intial annotation call

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6746)

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -1135,6 +1125,17 @@ func NewDaemon(dp datapath.Datapath) (*Daemon, *endpointRestoreState, error) {
 		log.Debugf("IPv6 health endpoint address: %s", health6)
 	}
 
+	if k8s.IsEnabled() {
```
Comment (aanm) [inline daemon/daemon.go:1156]: @tgraf add a small comment stating this needs to be done after all addresses are allocated

**Review by joestringer — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
Hit #6730

**Comment by tgraf** [comment]
test-me-please

Hit https://github.com/cilium/cilium/issues/6757

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**